ui-ngx/src/app/modules/home/pages/admin/oauth2/domains/domain-table-config.resolver.ts

```diff
@@ -84,16 +85,16 @@ export class DomainTableConfigResolver  {
     this.config.loadEntity = id => this.domainService.getDomainInfoById(id.id);
     this.config.saveEntity = (domain, originalDomain) => {
       const clientsIds = domain.oauth2ClientInfos as Array<string> || [];
-      let clientsTask: Observable<void>;
-      if (domain.id && !isEqual(domain.oauth2ClientInfos?.sort(),
-        originalDomain.oauth2ClientInfos?.map(info => info.id ? info.id.id : info).sort())) {
-        clientsTask = this.domainService.updateOauth2Clients(domain.id.id, clientsIds);
-      } else {
-        clientsTask = of(null);
-      }
       delete domain.oauth2ClientInfos;
-      return clientsTask.pipe(
-        mergeMap(() => this.domainService.saveDomain(domain, domain.id ? [] : clientsIds)),
+
+      return this.domainService.saveDomain(domain, domain.id ? [] : clientsIds).pipe(
+        switchMap(savedDomain => {
+          const shouldUpdateClients = domain.id && !isEqual(domain.oauth2ClientInfos?.sort(),
```

Comment:
The field `domain.oauth2ClientInfos` is always `undefined` because it is deleted from the `domain` object earlier in the code. To resolve this, review where the deletion occurs and ensure it aligns with the intended logic.